### ISSUE TITLE
Add automatic mixed precision

### ISSUE THREAD

**Issue by fredzzhang** [fredzzhang/pocket#35]
PyTorch 1.6 has incorporated automatic mixed precision (AMP), which was only available in APEX previously. Update the learning engines to apply AMP and possibly gradient clipping
